src/Attributes.jl

```diff
@@ -162,6 +162,27 @@ function has_attribute(G::Any, attr::Symbol)
    return D isa Dict && haskey(D, attr)
 end
 
+"""
+    get_attribute(f::Function, G::Any, attr::Symbol)
+
+Return the value stored for the attribute `attr`, or if no value has been set,
+return `f()`.
+
+This is intended to be called using `do` block syntax.
+
+```julia
+get_attribute(obj, attr) do
+    # default value calculated here if needed
+    ...
+end
+```
+"""
+function get_attribute(f, G::Any, attr::Symbol)
+   D = _get_attributes(G)
+   D isa Dict && return get(f, D, attr)
+   return f()
```

Comment:
Is there an example where `D` is not a dictionary?

Reply:
Yeah it can be `nothing`

Reply:
Ah right, thanks.